src/cmap/connection.ts

```diff
@@ -122,6 +123,8 @@ export interface ConnectionOptions
   extendedMetadata: Promise<Document>;
   /** @internal */
   mongoLogger?: MongoLogger | undefined;
+  /** @internal */
+  directConnection: boolean;
```

Comment:
Could we pass `directConnection` as an argument to `Connection.command()` from `Server.command()` instead of storing `directConnection` on the stream description, connection pool and (with a getter) on the Connection?

Reply:
alternatively, we already have some read preference logic in `Server.command()` (that's where we apply omitReadPreference) - perhaps we could just attach the read preference to commands there instead of in the connection class.

Reply:
With the exception of `loadBalanced` (which should not live on the stream description in my opinion), the stream description contains information about the connection.  `directConnection` is a client option that contains information about the topology (which is why it's stored on the topology).  In terms of code organization, it makes more sense to store this information on the topology.  

That also removes the need to store it on the connection pool, because we wouldn't need to assign it on the connection's description when the connection is created.

Reply:
To this point, I wish the pool/monitor was accessible from within a connection so we could reach up for this setting rather than passing it down

But I'm also fine with it on the description, the description represents a state which controls certain things about command construction (like if hello is ok) so controlling rp from it makes sense but I admit it is inconsistently deployed (à la ServerCapabilities topology.ts)

Reply:
But connection.ts does not have access to topology, only from the session, but the session does not always exist. You would need to do first what Neal said - refactor to access the pool/monitor from within a connection.

Reply:
Do we need to attach read preference to non-reads/writes (such as the authentication handshake)?  My thought is no - right?  

If so, the only places we need to attach read preference to commands are called from `Server.command()` - which does have access to the topology.

Reply:
I recall someone alluding to something about always attaching RP for simplicity, reads/writes etc all of the above so the driver doesn't need to discern between commands